HPC/README.md

```diff
@@ -0,0 +1,117 @@
+
+## Compilers
+Many HPC applications depend on compiler optimizations for better performance. We recommend the following compilers for HPC on Graviton instances
+* Gcc-10.2 or later version
+* Arm Compiler for Linux ([ACFL](https://developer.arm.com/Tools%20and%20Software/Arm%20Compiler%20for%20Linux))
+```
+#install ACFL
+sudo apt install environment-modules
+source /etc/profile.d/modules.sh
+#download acfl installer from Arm website
+tar xf arm-compiler-for-linux_22.1_Ubuntu-18.04_aarch64.tar
+sudo ./arm-compiler-for-linux_22.1_Ubuntu-18.04.sh -i /shared/arm
+module use /shared/arm/modulefiles
+```
+* Nvidia HPC SDK ([Nvhpc](https://developer.nvidia.com/hpc-sdk))
+```
+wget https://developer.download.nvidia.com/hpc-sdk/23.1/nvhpc_2023_231_Linux_aarch64_cuda_12.0.tar.gz
+tar xpzf nvhpc_2023_231_Linux_aarch64_cuda_12.0.tar.gz
+nvhpc_2023_231_Linux_aarch64_cuda_12.0/install
+module use /shared/nvidia/hpc_sdk/modulefiles
+```
+
+The recommended optimization flags for compilers are "-O3 -mcpu=native". The flag "-mcpu=native" can generate optimized code for a specific architecture including both Arm SVE and NEON. In addition, "--fast-math" for GCC and ACFL or "-fast" for Nvhpc can be used to enable fast math. The code generated by the three compilers have comparable performance. For Weather Research and Forcasting Model (WRF), binery generated by GCC-10.2 is 5% faster than ACFL and Nvhpc. If the application heavily depends on the performance of the transcendental functions (for example, BlackScholes used in Financial Services), we recommend ACFL because its capability to vectorize the transcendental functions and dramatically improve the performance.
+There are also options to turn on the vectorization report for performance tuning: "-fopt-info-vec" for GCC or "-Rpass=loop -Rpass-missed=loop -Rpass-analysis" for ACFL and NVHPC. 
+
+## Libraries
+We recommend to use [Arm Performance Libraries (Armpl)](https://developer.arm.com/Tools%20and%20Software/Arm%20Performance%20Libraries) because Armpl offers a vectorized math library (libamath), BLAS, LAPACK and FFT functions with best performance. Alternatively, [OpenBLAS](https://www.openblas.net/) and Nvidia BLAS library can be used as an alternative for linear algebra functions and [FFTW](http://www.fftw.org/) is a replacement for FFT functions.
+to use Armpl, 
+* with ACFL - compile with "-armpl" flag
+* with Gcc - link to the library with
+```
+module load <armpl>
+-I$ARMPL_INCLUDES -larmpl -L$ARMPL_LIBRARIES
+```
+
+We have noticed 15% better perfomance for HPL benchmark (dominated by DGEMM) on a hpc7g instance using Armpl than using OpenBlas and Nvidia Blas library. Below is instructions on how to link to different BLAS libraries.
```

Comment:
We have seen up to 15% better perfromance... so we recommend customers use ArmPL when they can.

ArmPL not Armpl.